### ISSUE TITLE
carrierwave manipulation, model.attribute nil in image uploader

### ISSUE THREAD

**Issue by geekdeepak** [carrierwaveuploader/carrierwave#1052]
``` ruby
process :scale
  #
  def scale
    manipulate! do |img|
      raise model.to_yaml
      if model.usage_tag == "logo"
        #raise model.to_yaml
        img.resize_to_fill!(320,320)
      elsif model.usage_tag == "thumbnail"
        img.resize_to_fill!(200,200)
      else
        img.resize_to_fill!(640,720)
      end
      img
    end
  end
```

the code above should manipulate the image through usage_tag, but the issue is raise model.to_yaml, model's attributes are nil. 

images are manipulated when i am updating them once object is created, 
i have 
location has many images and i am using images with fields_for of the location form.
and images are passing through this ImagesUploader.

**Comment by mixtmeta** [comment]
I just burned an hour debugging on this same issue.  Thanks @traviskochel for pointing out the cause of the bug.

**Comment by rafaeladel** [comment]
I was facing this problem, and wasted 5 hours trying to figure it out! All of that because the order of the strong parameters !!!!!

**Comment by gamezj** [comment]
Thanks @traviskochel

**Comment by fauzieuy** [comment]
Thanks @traviskochel, but how about with merge params, I have try but not working, an attribute still nil on uploader.

**Comment by norcali** [comment]
@traviskochel you sir are a godsend, I was going crazy with this

**Comment by feliperaul** [comment]
@bensie This was closed more then 4 years ago, but it still causing a lot of headaches (see posts above and also http://stackoverflow.com/a/23073723/1290457). 

Could you please reconsidering reopening this issue as a bug? 

In my tests, `some_model.images.create(file: image, some_other_attribute: anything)` makes `model.some_other_attribute` always return `nil` inside the uploader.

But if instead I do:

```
new_image = some_model.images.new
new_image.file = image
new_image.some_other_attribute = anything
```

then I can access `model.some_other_attribute` in the uploader.

**Comment by xhocquet** [comment]
I just ran into this issue today, luckily I found this post pretty quickly. Definitely still an issue with the most recent Carrierwave and Rails 5.1

Order of strong parameters prevents assigned attributes from being accessible in the Uploader.

**Comment by lpdahito** [comment]
Thanks @traviskochel you saved my day.

**Comment by damien-roche** [comment]
Severely disappointed CarrierWave. I've also reached for this lib over others. But this (which almost cost me an hour) and the fact it is a constant struggle to access the model or structure processing in any kind of sane way has me realising the massive mistake I made by using this lib in the first place.

To others, save yourself many headaches and go back to Paperclip, ActiveStorage (not much better), or Shrine: https://github.com/shrinerb/shrine
